UnitTests/GTMSessionFetcherFetchingTest.m

```diff
@@ -1783,28 +1835,81 @@ - (void)internalCancelFetchWithCallback:(unsigned int)sleepTime
                                                            parameters:@{@"sleep" : @"5"}];
   GTMSessionFetcher *fetcher = [self fetcherWithURLString:timeoutFileURLString];
   fetcher.stopFetchingTriggersCompletionHandler = YES;
-  if (authorizer) {
-    fetcher.authorizer = authorizer;
+
+  dispatch_semaphore_t authSemaphone = dispatch_semaphore_create(0);
+
+  if (sleepTime < 0) {
+    if (authorizerMode != TestAuthorizerModeNone) {
+      fetcher.authorizer = [TestAuthorizer asyncAuthorizerBlocked:^{
+        XCTFail(@"stopFetching called before begin should have prevented the authorizer from ever "
+                @"being called.");
+      }];
+    }
+  } else {
+    switch (authorizerMode) {
+      case TestAuthorizerModeNone:
+        break;
+      case TestAuthorizerModeSync:
+        fetcher.authorizer = [TestAuthorizer syncAuthorizer];
+        break;
+      case TestAuthorizerModeAsync:
+        fetcher.authorizer = [TestAuthorizer asyncAuthorizer];
+        break;
+      case TestAuthorizerModeWaitPreSleep:
+        XCTAssertGreaterThan(sleepTime, 0,
+                             @"Pre sleep when stop was before begin doensn't make sense.");
```

Comment:
"PreSleep when 'stop' was before 'begin' doesn't make sense."

Makes it easier to read if you put the quote in there

Reply:
Done